### ISSUE TITLE
AMP step() enforce synchronization

### ISSUE THREAD

**Issue by rBenke** [pytorch/pytorch#80167]
### 🐛 Describe the bug

# torch/cuda/amp/grad_scalar.py
def _maybe_opt_step(self, optimizer, optimizer_state, *args, **kwargs):
        retval = None
        if not sum(v.item() for v in optimizer_state["found_inf_per_device"].values()): # v is on gpu
            retval = optimizer.step(*args, **kwargs)
        return retval

### Versions

Collecting environment information...
PyTorch version: 1.11.0+cu115
Is debug build: False
CUDA used to build PyTorch: 11.5
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.4 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.31

Python version: 3.8.10 (default, Mar 15 2022, 12:22:08)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.4.0-105-generic-x86_64-with-glibc2.29
Is CUDA available: True
CUDA runtime version: 11.5.119
GPU models and configuration: GPU 0: NVIDIA A100 80GB PCIe
Nvidia driver version: 510.47.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.6
[pip3] torch==1.11.0+cu115
[pip3] torch-geometric==2.0.4
[pip3] torch-scatter==2.0.9
[pip3] torch-sparse==0.6.13
[conda] Could not collect

**Comment by ptrblck** [comment]
I think to avoid the synchronization you would need to use a fused optimizer, which provides the `_step_supports_amp_scaling` attribute (and can thus use [this code path](https://github.com/pytorch/pytorch/blob/648224dd805c11f5f4357dbf1d501be9c234b3db/torch/cuda/amp/grad_scaler.py#L324-L331) and check for invalid gradients internally).

CC @crcrpar who is working on fused optimizers.

**Comment by crcrpar** [comment]
currently we have an example in apex -- https://github.com/NVIDIA/apex/blob/master/apex/optimizers/fused_mixed_precision_lamb.py

**Comment by rBenke** [comment]
It works. Thanks!
